src/snowcli/cli/common/decorators.py

```diff
@@ -46,6 +47,18 @@ def global_options_with_connection(func: Callable):
     )
 
 
+def global_limited_options_with_connection(func: Callable):
+    """
+    Decorator providing default flags including connection flags for overriding
+    global parameters. Values are updated in global SnowCLI state.
+
+    To use this decorator your command needs to accept **options as last argument.
+    """
+    return _options_decorator_factory(
+        func, [*GLOBAL_CONNECTION_LIMITED_OPTIONS, *GLOBAL_OPTIONS]
+    )
+
+
```

Comment:
This change is introduced while we do not have the decorator that @sfc-gh-turbaszek discussed. 
It removes role and warehouse flags as those won't work in the native apps context as we have two separate roles that become active during `snow app run`. 
Alternatively, I can leave this out completely and just use the regular `global_options_with_connection` with a note in the docstring.

Reply:
Removed in agreement with @sfc-gh-turbaszek